### ISSUE TITLE
Please throw errors coming from wallet as they are

### ISSUE THREAD

**Issue by delivan** [apibara/starknet-react#543]
Currently, when `connect` is called and an error is returned from the wallet, a dapp receives only a `UserRejectedRequestError` object. However, the reasons for errors returned by the wallet are more diverse.

https://github.com/apibara/starknet-react/blob/b89a2be5f8f0070742033960554df8f3ea3d9370/packages/core/src/connectors/injected.ts#L129-L147

To allow the DApp to handle errors appropriately based on their specific cause, I propose that the wallet return the original error to a dapp.

**Comment by fracek** [comment]
I agree with this, but it will also be a breaking change. Will do that in v4.

**Comment by techpilot** [comment]
Could I grab this task?

**Comment by jaipaljadeja** [comment]
> [@jaipaljadeja](https://github.com/jaipaljadeja) We can also do this in v4.

Done!

**Comment by Simplyauf** [comment]
Can I work on this, please?
